pkg/config/template.go

```diff
@@ -47,3 +70,105 @@ func getTemplates(templatePath string) (CiTemplates, error) {
 func isYAML(file string, info os.FileInfo) bool {
 	return !info.IsDir() && (filepath.Ext(file) == ".yaml" || filepath.Ext(file) == ".yml")
 }
+
+// TemplateCMManager holds the details needed for the configmap controller
+type TemplateCMManager struct {
+	cmclient  corev1.ConfigMapInterface
+	prNumber  int
+	logger    *logrus.Entry
+	templates CiTemplates
+}
+
+// NewTemplateCMManager creates a new TemplateCMManager
+func NewTemplateCMManager(cmclient corev1.ConfigMapInterface, prNumber int, logger *logrus.Entry, templates CiTemplates) *TemplateCMManager {
+	return &TemplateCMManager{
+		cmclient:  cmclient,
+		prNumber:  prNumber,
+		logger:    logger,
+		templates: templates,
+	}
+}
+
+// CreateCMTemplates creates configMaps for all the changed templates.
+func (c *TemplateCMManager) CreateCMTemplates() error {
+	var errors []error
+	for filename, template := range c.templates {
+		templateData, err := GetTemplateData(template)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		cmName := GetTempCMName(template.Name, []string{filename, templateData})
+		cm := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cmName,
+				Labels: map[string]string{
+					createByRehearse:  "true",
+					rehearseLabelPull: strconv.Itoa(c.prNumber),
+				},
+			},
+			Data: map[string]string{filename: templateData},
+		}
+
+		c.logger.WithFields(logrus.Fields{"template-name": template.Name, "cm-name": cmName}).Info("creating rehearsal configMap for template")
+		if _, err := c.cmclient.Create(cm); err != nil {
+			if kerrors.IsAlreadyExists(err) {
+				if _, err := c.cmclient.Update(cm); err != nil {
```

Comment:
In theory we do not need to update, because we hash the content to the name, so the content is guaranteed to be identical, but I'd like to keep this - it makes the logic robust to the changes we could do to the names.